package.json

```diff
@@ -48,39 +49,42 @@
+    "lint:prettier:fix": "pnpm prettier -w 'app/assets/stylesheets/**/*.scss' 'app/assets/javascripts/**/*.{js,gjs,hbs}' 'plugins/**/assets/stylesheets/**/*.scss' 'plugins/**/assets/javascripts/**/*.{js,gjs,hbs}'",
     "lttf:ignore": "lint-to-the-future ignore",
     "lttf:output": "lint-to-the-future output -o ./lint-progress/",
-    "lint-progress": "yarn lttf:output && npx html-pages ./lint-progress --no-cache",
-    "postinstall": "./app/assets/javascripts/run-patch-package && rm -rf app/assets/javascripts/node_modules"
-  },
-  "workspaces": [
-    "app/assets/javascripts/admin",
-    "app/assets/javascripts/custom-proxy",
-    "app/assets/javascripts/deprecation-silencer",
-    "app/assets/javascripts/dialog-holder",
-    "app/assets/javascripts/discourse",
-    "app/assets/javascripts/discourse-common",
-    "app/assets/javascripts/discourse-hbr",
-    "app/assets/javascripts/discourse-i18n",
-    "app/assets/javascripts/discourse-markdown-it",
-    "app/assets/javascripts/discourse-plugins",
-    "app/assets/javascripts/discourse-widget-hbs",
-    "app/assets/javascripts/ember-cli-progress-ci",
-    "app/assets/javascripts/ember-production-deprecations",
-    "app/assets/javascripts/float-kit",
-    "app/assets/javascripts/pretty-text",
-    "app/assets/javascripts/select-kit",
-    "app/assets/javascripts/theme-transpiler",
-    "app/assets/javascripts/truth-helpers"
-  ],
-  "resolutions": {
-    "**/unset-value": "2.0.1"
+    "lint-progress": "pnpm lttf:output && npx html-pages ./lint-progress --no-cache",
+    "ember": "pnpm --dir=app/assets/javascripts/discourse ember"
   },
+  "packageManager": "pnpm@9.9.0",
   "engines": {
     "node": ">= 18",
-    "npm": "please-use-yarn",
-    "yarn": ">= 1.21.1"
+    "npm": "please-use-pnpm",
+    "yarn": "please-use-pnpm",
+    "pnpm": ">= 9"
+  },
+  "pnpm": {
+    "patchedDependencies": {
+      "ember-this-fallback@0.4.0": "patches/ember-this-fallback@0.4.0.patch",
+      "babel-plugin-debug-macros@0.3.4": "patches/babel-plugin-debug-macros@0.3.4.patch",
+      "content-tag@2.0.1": "patches/content-tag@2.0.1.patch",
+      "decorator-transforms@2.0.0": "patches/decorator-transforms@2.0.0.patch",
+      "virtual-dom@2.1.1": "patches/virtual-dom@2.1.1.patch",
+      "licensee@10.0.0": "patches/licensee@10.0.0.patch"
+    },
+    "peerDependencyRules": {
+      "allowedVersions": {
+        "@discourse/lint-configs>eslint": "*",
+        "@discourse/lint-configs>prettier": "*",
+        "lint-to-the-future-eslint>eslint": "*",
+        "@mixer/parallel-prettier>prettier": "*",
+        "lint-to-the-future-ember-template>ember-template-lint": "*",
+        "ember-this-fallback>ember-source": "*"
+      },
+      "allowAny": ["@uppy/*"],
+      "ignoreMissing": [
+        "webpack"
+      ]
+    }
```

Comment:
TIL

Reply:
Yeah, this is pnpm-specific stuff which allows us to silence all those warnings which yarn was spewing out 😅 

Some of them ideally need fixing properly (e.g. the uppy ones). But some (e.g. webpack) are happening because we have linting/dev dependencies in the root package.json, which have peerdeps on `ember-source`. And... we don't want ember-source in our root package.json.